### PR TITLE
Ensure that incremental builds work with 'gulp build'.

### PR DESCRIPTION
**Pull request by loganfsmyth**
| Q                        | A 
| ------------------------ | ---
| Patch: Bug Fix?          | N
| Major: Breaking Change?  | N
| Minor: New Feature?      | N
| Deprecations?            | 
| Spec Compliancy?         | 
| Tests Added/Pass?        | 
| Fixed Tickets            |  
| License                  | MIT
| Doc PR                   | 
| Dependency Changes       |

### CODE REVIEW

File: Gulpfile.js [hunk 1]
```diff
@@ -10,41 +10,39 @@ const gutil = require("gulp-util");
...
 const scripts = "./packages/*/src/**/*.js";
-let srcEx, libFragment;
 
-if (path.win32 === path) {
-  srcEx = /(packages\\[^\\]+)\\src\\/;
-  libFragment = "$1\\lib\\";
-} else {
-  srcEx = new RegExp("(packages/[^/]+)/src/");
-  libFragment = "$1/lib/";
+function swapSrcWithLib(srcPath) {
```
Comment (loganfsmyth) [inline Gulpfile.js]: These regexes were gross, so figured this would be easier to follow.

File: Gulpfile.js [hunk 2]
```diff
@@ -10,41 +10,39 @@ const gutil = require("gulp-util");
...
 }
 
-const mapToDest = function (path) { return path.replace(srcEx, libFragment); };
-const dest = "packages";
-
 gulp.task("default", ["build"]);
 
 gulp.task("build", function () {
-  return gulp.src(scripts)
+  return gulp.src(scripts, { base: base })
```
Comment (loganfsmyth) [inline Gulpfile.js]: This was the core of the problem, we weren't setting a base, so all the filepaths were `babel-core/src/index.js`, and the regexes all expected the path to start with `packages`.

File: Gulpfile.js [hunk 3]
```diff
@@ -10,41 +10,39 @@ const gutil = require("gulp-util");
...
       },
     }))
-    .pipe(newer({ map: mapToDest }))
+    .pipe(newer({
+      dest: base,
+      map: swapSrcWithLib,
+    }))
     .pipe(through.obj(function (file, enc, callback) {
-      gutil.log("Compiling", "'" + chalk.cyan(file.path) + "'...");
+      gutil.log("Compiling", "'" + chalk.cyan(file.relative) + "'...");
```
Comment (loganfsmyth) [inline Gulpfile.js]: This I just changed since it seems easier to read.

**Comment by existentialism** [comment]
Nice work!

**Pull request merged**